### PR TITLE
feat: pause e2e tests for fiattokenfactory

### PR DESCRIPTION
**Pull request by boojamya**
Includes:
- `TestFiatTFUpdatePauser`
- `TestFiatTFPause`
- `TestFiatTFUnpause`

### CODE REVIEW

File: e2e/fiat_tf_test.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+	showPauserRes, err = showPauser(ctx, val)
+	require.NoError(t, err, "failed to query show-pauser")
+	expectedGetPauserResponse = fiattokenfactorytypes.QueryGetPauserResponse{
+		Pauser: fiattokenfactorytypes.Pauser{
+			Address: string(newPauser3.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetPauserResponse.Pauser, showPauserRes.Pauser)
+}
```
Comment (xngln) [inline e2e/fiat_tf_test.go:105]: Could we add the happy path case as well? (i.e. a normal request — nothing is paused or blacklisted)

File: e2e/fiat_tf_test.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	showPausedRes, err := showPaused(ctx, val)
+	require.NoError(t, err, "error querying for paused state")
+	expectedPaused := fiattokenfactorytypes.QueryGetPausedResponse{
+		Paused: fiattokenfactorytypes.Paused{
+			Paused: false,
+		},
+	}
+	require.Equal(t, expectedPaused, showPausedRes)
+
+	// ACTION: Pause TF from a blacklisted Pauser account
```
Comment (xngln) [inline e2e/fiat_tf_test.go:137]: Seems that assertions of the expected result are missing from this case and the one below?

File: e2e/fiat_tf_test.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Pause TF while TF is already paused
+	// EXPECTED: Success; TF remains paused
+	// Status:
+	// 	Paused: true
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+}
```
Comment (xngln) [inline e2e/fiat_tf_test.go:154]: Can we add a happy path test here? (precondition: valid pause request while nothing is paused or blacklisted)

File: e2e/fiat_tf_test.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	// Status:
+	// 	Paused: true
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Unpause TF while TF is already paused
```
Comment (xngln) [inline e2e/fiat_tf_test.go:199]: Typo here :) `is already paused`, but precondition is that TF is not paused

File: e2e/fiat_tf_test.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+	showPausedRes, err := showPaused(ctx, val)
+	require.NoError(t, err, "error querying for paused state")
+	expectedPaused := fiattokenfactorytypes.QueryGetPausedResponse{
+		Paused: fiattokenfactorytypes.Paused{
+			Paused: true,
+		},
+	}
+	require.Equal(t, expectedPaused, showPausedRes)
+
+	// ACTION: Unpause TF from a blacklisted Pauser account
```
Comment (xngln) [inline e2e/fiat_tf_test.go:188]: Missing assertions of expected result for this test and the one below

File: e2e/fiat_tf_test.go [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+	// ACTION: Unpause TF while TF is already paused
+	// EXPECTED: Success; TF remains unpaused
+	// Status:
+	// 	Paused: false
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
```
Comment (xngln) [inline e2e/fiat_tf_test.go:208]: Duplicate `unpause`?

File: e2e/fiat_tf_test.go [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+}
```
Comment (xngln) [inline e2e/fiat_tf_test.go:209]: Can we get a happy path test here as well? So unpausing a paused TF with an unblacklisted pauser.

File: e2e/fiat_tf_test.go (hunk 1 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:105]: Added in https://github.com/noble-assets/noble/pull/368/commits/a278e1e59be0319c50ef61a3a1ff79105c8fa702

File: e2e/fiat_tf_test.go (hunk 3 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:154]: Added in https://github.com/noble-assets/noble/pull/368/commits/a278e1e59be0319c50ef61a3a1ff79105c8fa702

File: e2e/fiat_tf_test.go (hunk 2 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:137]: We do an assertion inside the `pauseFiatTF` function: https://github.com/noble-assets/noble/blob/149a7f6f8748b761bc2ce2108da09c27916b101a/e2e/utils.go#L421-L433

File: e2e/fiat_tf_test.go (hunk 5 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:188]: We do this assertion in the `unpauseFiatTF` function: https://github.com/noble-assets/noble/blob/149a7f6f8748b761bc2ce2108da09c27916b101a/e2e/utils.go#L445-L454

File: e2e/fiat_tf_test.go (hunk 4 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go]: Updated! [`2910fdb` (#369)](https://github.com/noble-assets/noble/pull/369/commits/2910fdb9d6ee5da808061f8f7c7e005a04f64e75)

File: e2e/fiat_tf_test.go (hunk 1 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:125]: Added in: [`9a77cc4` (#369)](https://github.com/noble-assets/noble/pull/369/commits/9a77cc480b7a8b9d0534414f76a3f8325109702d)

File: e2e/fiat_tf_test.go (hunk 3 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:183]: Added in: [`9a77cc4` (#369)](https://github.com/noble-assets/noble/pull/369/commits/9a77cc480b7a8b9d0534414f76a3f8325109702d)

File: e2e/fiat_tf_test.go (hunk 7 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:241]: Added in: [`9a77cc4` (#369)](https://github.com/noble-assets/noble/pull/369/commits/9a77cc480b7a8b9d0534414f76a3f8325109702d)

**Review by kaliweh-circle — APPROVED** [review]

**Review by johnletey — APPROVED** [review]
LGTM! Passing on Cosmos SDK v0.50 ⚡

- ✅ Green CI for `TestFiatTFUpdatePauser` ([link](https://github.com/noble-assets/noble/actions/runs/11381745994/job/31663982801?pr=369))
- ✅ Green CI for `TestFiatTFPause` ([link](https://github.com/noble-assets/noble/actions/runs/11381745994/job/31663983108?pr=369))
- ✅ Green CI for `TestFiatTFUnpause` ([link](https://github.com/noble-assets/noble/actions/runs/11381745994/job/31663983457?pr=369))

**Pull request merged**